File: lib/src/model/hms_subscribe_degradation_params.dart
///100ms HMSSubscribeDegradationParams
///
///[HMSSubscribeDegradationParams] contains degradeGracePeriodSeconds, packetLossThreshold and recoverGracePeriodSeconds.

class HMSSubscribeDegradationParams {
  final int degradeGracePeriodSeconds;
  final int packetLossThreshold;
  final int recoverGracePeriodSeconds;
  HMSSubscribeDegradationParams(
      {required this.degradeGracePeriodSeconds,
      required this.packetLossThreshold,
      required this.recoverGracePeriodSeconds});
  factory HMSSubscribeDegradationParams.fromMap(Map map) {
    return HMSSubscribeDegradationParams(
        degradeGracePeriodSeconds: map["degrade_grace_period_seconds"].toInt(),
        packetLossThreshold: map["packet_loss_threshold"].toInt(),
        recoverGracePeriodSeconds: map["recover_grace_period_seconds"].toInt());
  }
}

File: packages/hmssdk_flutter/android/src/main/kotlin/live/hms/hmssdk_flutter/HMSSubscribeDegradationParams.kt
import live.hms.video.sdk.models.role.SubscribeDegradationParams

class HMSSubscribeDegradationParams {
    companion object {
        fun toDictionary(subscribeDegradationParams: SubscribeDegradationParams): HashMap<String, Any> {
            val args = HashMap<String, Any>()
            args["degrade_grace_period_seconds"] = subscribeDegradationParams.degradeGracePeriodSeconds
            args["packet_loss_threshold"] = subscribeDegradationParams.packetLossThreshold
            args["recover_grace_period_seconds"] = subscribeDegradationParams.recoverGracePeriodSeconds
            return args
        }
    }
}
